src/crimson/osd/backfill_state.cc

```diff
@@ -327,15 +327,24 @@ BackfillState::Enqueuing::Enqueuing(my_context ctx)
   }
   trim_backfill_infos();
 
+  if (should_rescan_primary(backfill_state().peer_backfill_info,
+				   primary_bi)) {
+    // need to grab one another chunk of the object namespace and restart
+    // the queueing.
+    logger().debug("{}: reached end for current local chunk", __func__);
+    post_event(RequestPrimaryScanning{});
+    return;
+  } else if (should_rescan_replicas(backfill_state().peer_backfill_info,
+				    primary_bi)) {
+    // Count simultaneous scans as a single op and let those complete
+    post_event(RequestReplicasScanning{});
+    return;
+  }
```

Comment:
This works because we now always loop back to Enqueuing, right?  Seems cleaner.

Reply:
Yes, that's right.